fix: event Duration displays incorrectly in event name

divyanshchahar opened the pull request:
## What does this PR do?

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

- Fixes #16887
 
[Loom video of the fix](https://www.loom.com/share/0d01be1f663f45e3920645e2a503d41c?sid=c5c0ac1b-7dec-43c6-b9be-b6aa49b63389)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

Steps to test the fix (see loom video)
1. Create an event with the following attributes
1.1. event has multiple duration
1.2. booker is allowed to select duration
1.3 under advanced tab of the event the `with event name` field is setup something like this `Event Name {Event duration}: {Organiser} and {Scheduler}`
2. Book this event with any thing but the default duration
3. Check `baseaddress/booking/{bookingid}` 
4. The name of the event **should include the selected duration**
5. The email **should include the selected duration**

___

- Are there environment variables that should be set ?  **no**
- What are the minimal test data to have? A regular user and an event as described above
- What is expected (happy path) to have (input and output)? ** name of the event is in line with selected non default duration everywhere**
- Any other important info that could help to test that PR ? **no**

## Checklist

## What caused the bug 

`getEventName` function in `./packages/core/event.ts` was computing event name based on the default length of the event

## How was the issue fixed ?
- `getEventName` was updated to accept two additional optional parameters (start time and end time) and if these parameters were passed then the duration in the event name is computed from these two additional parameters instead of using the default value

- function calls to the above function were updated to pass the two new arguments

## Environment

production

divyanshchahar committed:
updated function

divyanshchahar committed:
updated function call

divyanshchahar committed:
updated function call

divyanshchahar committed:
updated function call

divyanshchahar committed:
Merge branch 'main' into fix#16887

anikdhabal commented on packages/core/event.ts:
```diff
@@ -53,6 +59,11 @@ export function getEventName(eventNameObj: EventNameObjectType, forAttendeeView
     eventName = eventName.replace("{LOCATION}", locationString);
   }
 
+  const duration =
+    startTime && endTime
+      ? `${dayjs(endTime).diff(dayjs(startTime), "minute")} mins`
+      : `${String(eventNameObj.eventDuration)} mins`;
+
```
We are not fixing the actual bug; instead, we are adding a solution on top of it just to show the event duration. I don't think this is a good way to fix it. It used to work before. Could you check that more deeply?

anikdhabal reviewed (changes requested):
@divyanshchahar thanks for your work. Left a comment

divyanshchahar committed:
updated title atttribute
- earlier title attribute was calcullated using a function
- now this attribute is directly read from calEvent object

divyanshchahar committed:
updated eventName variable
- earlier this variable was calculated using a function
- now this variable is read from bookingInfo object

divyanshchahar committed:
Merge branch 'main' into fix#16887

divyanshchahar commented:
@anikdhabal can you please provide some feedback on my latest comments

anikdhabal commented:
@divyanshchahar thanks for your work. but I think it was fixed.  Let me know if you can reproduce it
